<?xml version="1.0"?>
<launch>

  <node name="toppra_trajectory_generator" pkg="topp_ros" type="generate_toppra_trajectory.py"/>
  
  <param name="room_coordinates_file_UAV1" value="$(find pomdp_hrs_planner)/config/room_coordinates.yaml"/>
  
  <remap from="room_coordinates_file" to="room_coordinates_file_UAV1"/>
  <remap from="pose" to="UAV1/pose"/>
  <remap from="action" to="UAV1/action"/>
  <remap from="multi_dof_trajectory" to="UAV1/multi_dof_trajectory"/>
  <node name="building_tour1" pkg="pomdp_hrs_planner" type="waypoints.py" output="screen"/>
  
  <param name="room_coordinates_file_UAV2" value="$(find pomdp_hrs_planner)/config/room_coordinates_2.yaml"/>

  <remap from="room_coordinates_file" to="room_coordinates_file_UAV2"/>
  <remap from="pose" to="UAV2/pose"/>
  <remap from="action" to="UAV2/action"/>
  <remap from="multi_dof_trajectory" to="UAV2/multi_dof_trajectory"/>
  <node name="building_tour2" pkg="pomdp_hrs_planner" type="waypoints.py" output="screen"/>
  
  <param name="human_loc" value="room3"/>
  <param name="fire_loc" value="room1"/>
  <param name="team_config_file" value="$(find pomdp_hrs_planner)/config/team_abilities_2.yaml"/>
  <param name="obs_config_file" value="$(find pomdp_hrs_planner)/config/obs_config.yaml"/>

  	<group ns="UAV1">
 	<include file="$(find pomdp_hrs_planner)/launch/single_pomdp.launch">
 	     <arg name="pomdp_model" value="$(find pomdp_hrs_planner)/config/agent1_final.pomdpx" />
     	<arg name="pomdp_policy" value="$(find pomdp_hrs_planner)/config/agent1_final.policyx" />
     	<arg name="consensus" value="0" />
	</include>
	</group>
  	<group ns="UAV2">
 	<include file="$(find pomdp_hrs_planner)/launch/single_pomdp.launch">
 		<arg name="pomdp_model" value="$(find pomdp_hrs_planner)/config/agent2_final.pomdpx" />
     	<arg name="pomdp_policy" value="$(find pomdp_hrs_planner)/config/agent2_final.policyx" />
     	<arg name="consensus" value="0" />
	</include>
	</group>
	<node name="observer" pkg="pomdp_hrs_planner" type="fire_observation_generator.py" output="screen">
        <param name="room" value="fire"/>
  </node>
  
	<!-- <include file="$(find larics_gazebo_worlds)/launch/floor_plan_simple.launch">
		<arg name="args" value=""/>
	</include>

	 <include file="$(find mmuav_description)/launch/spawn_uav.launch">
    <arg name="model_type" value="uav" />
    <arg name="name" value="UAV1" />
    <arg name="x" value="5" />
    <arg name="y" value="-12" />
  </include>

  <include file="$(find mmuav_control)/launch/rotors_variation_attitude_position_control.launch">
    <arg name="namespace" value="UAV1" />
    <arg name="x" value="5" />
    <arg name="y" value="-12" />
  </include> -->

  <!--<include file="$(find mmuav_description)/launch/spawn_uav.launch">
    <arg name="model_type" value="uav" />
    <arg name="name" value="UAV2" />
    <arg name="x" value="5" />
    <arg name="y" value="-13" />
  </include>

  <include file="$(find mmuav_control)/launch/rotors_variation_attitude_position_control.launch">
    <arg name="namespace" value="UAV2" />
    <arg name="x" value="5" />
    <arg name="y" value="-13" />
  </include>-->
	
</launch>
